tests/functional/tests/mds/test_mds.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+
+class TestMDSs(object):
+
+    @pytest.mark.no_docker
+    def test_mds_is_installed(self, node, host):
+        assert host.package("ceph-mds").is_installed
+
+    def test_mds_service_is_running(self, node, host):
+        service_name = "ceph-mds@{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert host.service(service_name).is_running
+
+    def test_mds_service_is_enabled(self, node, host):
+        service_name = "ceph-mds@{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert host.service(service_name).is_enabled
+
+    @pytest.mark.no_docker
+    def test_mds_is_up(self, node, host):
+        cmd = "sudo ceph --name client.bootstrap-mds --keyring /var/lib/ceph/bootstrap-mds/{cluster}.keyring --cluster={cluster} --connect-timeout 5 -f json -s | python -c 'import sys, json; print(json.load(sys.stdin)[\"fsmap\"][\"by_rank\"][0][\"name\"])'".format(cluster=node["cluster_name"])
+        output = host.check_output(cmd)
+        phrase = "{hostname}".format(hostname=node["vars"]["inventory_hostname"])
+        assert phrase in output
+
+    @pytest.mark.docker
+    def test_mds_is_up(self, node, host):
```

Comment:
`test_mds_is_up()` is redefined here, it means it will never be tested on non-containerized deployment.